src/CoreWCF.RabbitMQ.Client/src/CoreWCF/ServiceModel/Channels/RabbitMqBinding.cs

```diff
@@ -0,0 +1,117 @@
+﻿using System;
+using System.Net;
+using RabbitMQ.Client;
+using BinaryMessageEncodingBindingElement = System.ServiceModel.Channels.BinaryMessageEncodingBindingElement;
+using Binding = System.ServiceModel.Channels.Binding;
+using BindingElementCollection = System.ServiceModel.Channels.BindingElementCollection;
+using TextMessageEncodingBindingElement = System.ServiceModel.Channels.TextMessageEncodingBindingElement;
+
+namespace CoreWCF.ServiceModel.Channels
+{
+    public class RabbitMqBinding : Binding
+    {
+        private RabbitMqTransportBindingElement _transport;
+        private TextMessageEncodingBindingElement _textMessageEncodingBindingElement;
+        private BinaryMessageEncodingBindingElement _binaryMessageEncodingBindingElement;
+        private RabbitMqMessageEncoding _messageEncoding = RabbitMqMessageEncoding.Text;
+
+        public RabbitMqBinding(string uri) : this(new Uri(uri))
+        { }
+
+        public RabbitMqBinding(Uri uri)
+        {
```

Comment:
You don't normally pass the Uri to the binding. You pass it to the ChannelFactory constructor and you get told the Uri in RabbitMqChannelFactory.OnCreateChannel